PhysicsTools/JetMCAlgos/python/HadronAndPartonSelector_cfi.py

```diff
@@ -5,8 +5,12 @@
     src = cms.InputTag("generator"),
     particles = cms.InputTag("genParticles"),
     partonMode = cms.string("Auto"),
+    CheckHerwig7Flag = cms.bool(False),
     fullChainPhysPartons = cms.bool(True)
 )
 # select hadrons and partons for the slimmedGenJetsFlavourInfos, required for origin identification
 
 selectedHadronsAndPartonsForGenJetsFlavourInfos = selectedHadronsAndPartons.clone(particles = "prunedGenParticles")
+
+from Configuration.Eras.Modifier_run2_JMENanoHerwig7_cff import Modifier_run2_JMENanoHerwig7
```

Comment:
this should be done inside your JMEnano configuration ! w/o polluting the general cms-sw
PhysicsTools/NanoAOD/python/custom_jme_cff.py